### PR TITLE
smart: fix handling for controller

### LINKED ISSUE #4881: SMART does not found device behind Raid-Controller
### Relevant telegraf.conf:

### System info:

[Include Telegraf version, operating system name, and other relevant details]

### Steps to reproduce:

```
/dev/bus/0 -d megaraid,6 # /dev/bus/0 [megaraid_disk_06], SCSI device
```

### Expected behavior:

Strip device by first `#`.
Result: `/dev/bus/0 -d megaraid,6`

### Actual behavior:

Strip device by first space.
Result: `/dev/bus/0`

### Additional info:

```
4 root@pavo ~ # smartctl --scan                                                                                                                                                                                                                                                :(
/dev/sda -d scsi # /dev/sda, SCSI device
/dev/bus/0 -d megaraid,5 # /dev/bus/0 [megaraid_disk_05], SCSI device
/dev/bus/0 -d megaraid,6 # /dev/bus/0 [megaraid_disk_06], SCSI device
root@pavo ~ # /usr/bin/smartctl --info --health --attributes --tolerance=verypermissive -n standby --format=brief /dev/bus/0               
smartctl 6.6 2017-11-05 r4594 [x86_64-linux-4.18.14-arch1-1-ARCH] (local build)
Copyright (C) 2002-17, Bruce Allen, Christian Franke, www.smartmontools.org

/dev/bus/0: Unable to detect device type
Please specify device type with the -d option.

Use smartctl -h to get a usage summary

1 root@pavo ~ # /usr/bin/smartctl --info --health --attributes --tolerance=verypermissive -n standby --format=brief -d megaraid,5  /dev/bus/0                                                                                                                                  :(
smartctl 6.6 2017-11-05 r4594 [x86_64-linux-4.18.14-arch1-1-ARCH] (local build)
Copyright (C) 2002-17, Bruce Allen, Christian Franke, www.smartmontools.org

CHECK POWER MODE not implemented, ignoring -n option
=== START OF INFORMATION SECTION ===
Model Family:     Seagate Constellation ES.2 (SATA 6Gb/s)
```

### PR DESCRIPTION
**Pull request by genofire**
fix #4881 

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/smart/smart.go [hunk 1]
```diff
@@ -198,8 +198,12 @@ func gatherDisk(acc telegraf.Accumulator, usesudo, attributes bool, smartctl, no
 	}
 
 	device_tags := map[string]string{}
-	device_node := strings.Split(device, " ")[0]
-	device_tags["device"] = path.Base(device_node)
+	device_node := strings.SplitN(device, " ", 2)
+	if device_node[1] == "" {
```
Comment (glinton) [inline plugins/inputs/smart/smart.go]: Make this `if len(device_node) < 2` to avoid a panic

**Comment by genofire** [comment]
@glinton sry, i have no idea how to test it

File: plugins/inputs/smart/smart.go [hunk 2]
```diff
@@ -136,7 +136,7 @@ func (m *Smart) scan() ([]string, error) {
 
 	devices := []string{}
 	for _, line := range strings.Split(string(out), "\n") {
-		dev := strings.Split(line, " ")
```
Comment (glinton) [inline plugins/inputs/smart/smart.go]: Although it is more overhead, this should be:
```go
		dev := []string{}
		if strings.Contains(line, "#") {
			dev = strings.Split(line, "#")
		} else {
			dev = strings.Split(line, " ")
		}
```
To handle any case where `smartctl --scan` returns something like:
```
/dev/sda -d scsi
/dev/sdb -d scsi
```

**Comment by glinton** [comment]
I'm starting to wonder if this is the best method of solving this problem as always appending the `-d type` yields different results:
```
> sudo smartctl --scan
/dev/sda -d scsi # /dev/sda, SCSI device
```
```
> sudo smartctl --info --health --attributes --tolerance=verypermissive -n standby --format=brief /dev/sda -d scsi
smartctl 6.6 2017-11-05 r4594 [x86_64-linux-5.0.0-13-generic] (local build)
Copyright (C) 2002-17, Bruce Allen, Christian Franke, www.smartmontools.org

User Capacity:        250,059,350,016 bytes [250 GB]
Logical block size:   512 bytes
LU is fully provisioned
Rotation Rate:        7200 rpm
Logical Unit id:      0x5000c5003e2a977e
Serial number:        5VY6DBY3
Device type:          disk
Local Time is:        Thu Apr 25 09:39:19 2019 MDT
SMART support is:     Unavailable - device lacks SMART capability.

=== START OF READ SMART DATA SECTION ===
Current Drive Temperature:     0 C
Drive Trip Temperature:        0 C
```
```
> sudo smartctl --info --health --attributes --tolerance=verypermissive -n standby --format=brief /dev/sda 
smartctl 6.6 2017-11-05 r4594 [x86_64-linux-5.0.0-13-generic] (local build)
Copyright (C) 2002-17, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Model Family:     Seagate Barracuda 7200.12
Device Model:     ST3250318AS
Serial Number:    5VY6DBY3
LU WWN Device Id: 5 000c50 03e2a977e
Firmware Version: CC6B
User Capacity:    250,059,350,016 bytes [250 GB]
Sector Size:      512 bytes logical/physical
Rotation Rate:    7200 rpm
Device is:        In smartctl database [for details use: -P show]
ATA Version is:   ATA8-ACS T13/1699-D revision 4
SATA Version is:  SATA 2.6, 3.0 Gb/s (current: 3.0 Gb/s)
Local Time is:    Thu Apr 25 09:39:23 2019 MDT

==> WARNING: A firmware update for this drive may be available,
see the following Seagate web pages:
http://knowledge.seagate.com/articles/en_US/FAQ/207931en
http://knowledge.seagate.com/articles/en_US/FAQ/213891en

SMART support is: Available - device has SMART capability.
SMART support is: Enabled
Power mode is:    ACTIVE or IDLE

=== START OF READ SMART DATA SECTION ===
SMART overall-health self-assessment test result: PASSED

SMART Attributes Data Structure revision number: 10
Vendor Specific SMART Attributes with Thresholds:
ID# ATTRIBUTE_NAME          FLAGS    VALUE WORST THRESH FAIL RAW_VALUE
  1 Raw_Read_Error_Rate     POSR--   110   099   006    -    208301221
  3 Spin_Up_Time            PO----   097   097   000    -    0
  4 Start_Stop_Count        -O--CK   100   100   020    -    823
  5 Reallocated_Sector_Ct   PO--CK   100   100   036    -    0
  7 Seek_Error_Rate         POSR--   085   060   030    -    382409253
  9 Power_On_Hours          -O--CK   054   054   000    -    41171
 10 Spin_Retry_Count        PO--C-   100   100   097    -    0
 12 Power_Cycle_Count       -O--CK   100   100   020    -    410
183 Runtime_Bad_Block       -O--CK   100   100   000    -    0
184 End-to-End_Error        -O--CK   100   100   099    -    0
187 Reported_Uncorrect      -O--CK   100   100   000    -    0
188 Command_Timeout         -O--CK   100   098   000    -    12885164044
189 High_Fly_Writes         -O-RCK   100   100   000    -    0
190 Airflow_Temperature_Cel -O---K   065   049   045    -    35 (Min/Max 18/37)
194 Temperature_Celsius     -O---K   035   051   000    -    35 (0 12 0 0 0)
195 Hardware_ECC_Recovered  -O-RC-   039   018   000    -    208301221
197 Current_Pending_Sector  -O--C-   100   100   000    -    0
198 Offline_Uncorrectable   ----C-   100   100   000    -    0
199 UDMA_CRC_Error_Count    -OSRCK   200   200   000    -    0
240 Head_Flying_Hours       ------   100   253   000    -    42184 (170 229 0)
241 Total_LBAs_Written      ------   100   253   000    -    4110008082
242 Total_LBAs_Read         ------   100   253   000    -    2199341897
                            ||||||_ K auto-keep
                            |||||__ C event count
                            ||||___ R error rate
                            |||____ S speed/performance
                            ||_____ O updated online
                            |______ P prefailure warning

```

**Pull request closed without merging**